### PR TITLE
mds: convert stringstream to CachedStackStringStream

### PR DESCRIPTION
**Pull request by batrick**
This is a simple performance refactor.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
ping @kotreshhr @mchangir

File: src/mds/SessionMap.cc [hunk 1]
```diff
@@ -855,15 +855,15 @@ void SessionMap::save_if_dirty(const std::set<entity_name_t> &tgt_sessions,
...
+    CachedStackStringStream css;
+    *css << session_id;
 
     // Serialize V
     bufferlist bl;
     session->info.encode(bl, mds->mdsmap->get_up_features());
 
     // Add to RADOS op
-    to_set[k.str()] = bl;
+    to_set[std::string(css->strv())] = bl;
```
Comment (kotreshhr) [inline src/mds/SessionMap.cc:866]: nit: css->str()

**Review by kotreshhr — APPROVED** [review]

File: src/mds/SessionMap.cc (hunk 1 above)
Comment (batrick) [reply src/mds/SessionMap.cc:866]: fixed, thanks!

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-09-18_04:24:45-fs-wip-pdonnell-testing-20200918.022351-distro-basic-smithi/

failures unrelated

**Pull request merged**